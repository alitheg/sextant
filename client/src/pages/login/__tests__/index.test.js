// Link.react.test.js
import React from 'react';
import LoginPage from '..';
import {
  BrowserRouter as Router
} from "react-router-dom";
import renderer from 'react-test-renderer';
import {cleanup, render} from '@testing-library/react';

afterEach(cleanup);

describe('Login Page', () => {
  test('Matches snapshots', () => {
    let component = renderer.create(
      <Router>
        <LoginPage loggedIn={false} />,
      </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
      <Router>
        <LoginPage loggedIn={true} />,
      </Router>
    );

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  test('When not logged in, renders form', () => {
    const { getByLabelText } = render(
      <Router>
        <LoginPage loggedIn={false} />,
      </Router>
    );

    expect(getByLabelText(/Username/i)).toBeTruthy()
    expect(getByLabelText(/Password/i)).toBeTruthy()
  })
  test('When logged in, renders redirect', () => {
    const { queryByLabelText } = render(
      <Router>
        <LoginPage loggedIn={true} />,
      </Router>
    );

    expect(queryByLabelText(/Username/i)).toBeNull()
    expect(queryByLabelText(/Password/i)).toBeNull()
  })
})
