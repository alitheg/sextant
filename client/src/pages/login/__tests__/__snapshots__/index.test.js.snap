// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Matches snapshots 1`] = `
Array [
  <div
    className="container"
  >
    <h1>
      Log In
    </h1>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="userUpdate.Username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="userUpdate.Username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="string"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="userUpdate.Password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="userUpdate.Password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </form>
  </div>,
  ",",
]
`;

exports[`Login Page Matches snapshots 2`] = `","`;
